cradlepoint/pages/api/editEngagement.js

```diff
@@ -13,15 +13,16 @@ export default async (req, res) => {
     const data = req.body;
     const valid = await engagementSchema.isValid(data);
     if (valid && ObjectId.isValid(data.testPlanId)){
-      const result = engagementSchema.cast(data);
+      const validData = engagementSchema.cast(data);
       // Set ID strings to Mongo ObjectId's
-      const id = ObjectId(result._id);
-      const testplanId = ObjectId(result.testPlanId);
+      const id = ObjectId(validData._id);
+      const testplanId = ObjectId(validData.testPlanId);
+      // Create the database query and replacement object
       const query = {_id: id};
-      const engagement = {...result , _id: id, testPlanId:testplanId};
+      const newEngagement = {...validData , _id: id, testPlanId:testplanId};
       // Update the Database w/ new Engagement
       const db = await connectToDb();
-      await db.collection("engagements").replaceOne(query, engagement);
+      await db.collection("engagements").replaceOne(query, newEngagement);
```

Comment:
This endpoint isn't working for me. Changing just the objectId, I get input not in right format

Reply:
Tried exactly that, and it worked for me!